### PR TITLE
Make compiler.properties fall back to prefixed

### PR DESCRIPTION
**Pull request by puffnfresh**
Previously, if we wanted to override the shell.prompt property, we had
to modify compiler.properties in the jar. This change lets us do the
following, instead:

```
scala -Dscala.shell.prompt="$(echo -e "\npuffnfresh> ")"
```

All properties previously loaded from compiler.properties now fall back
to "scala." in the system properties when not found.

### CODE REVIEW

**Comment by retronym** [comment]
Here are the configuration properties currently loaded via this mechanism.

```
def residentPromptString = scalaPropOrElse("resident.prompt", "\nnsc> ")
def shellPromptString    = scalaPropOrElse("shell.prompt", "\nscala> ")
def shellInterruptedString = scalaPropOrElse("shell.interrupted", ":quit\n")

val versionString         = "version " + scalaPropOrElse("version.number", "(unknown)")
val copyrightString       = scalaPropOrElse("copyright.string", "Copyright 2002-2013, LAMP/EPFL")

/** This is the encoding to use reading in source files, overridden with -encoding.
 *  Note that it uses "prop" i.e. looks in the scala jar, not the system properties.
 */
def sourceEncoding        = scalaPropOrElse("file.encoding", "UTF-8")
def sourceReader          = scalaPropOrElse("source.reader", "scala.tools.nsc.io.SourceReader")

/** Either the development or release version if known, otherwise
 *  the empty string.
 */
def versionNumberString = scalaPropOrEmpty("version.number")

v <- scalaPropOrNone("maven.version.number")

ov <- scalaPropOrNone("version.number")
```

The chance of an accidental collision with a JVM property in someone's environment seems low enough to me.

File: src/library/scala/util/Properties.scala [hunk 1]
```diff
@@ -63,9 +63,9 @@ private[scala] trait PropertiesTrait {
   def envOrSome(name: String, alt: Option[String])       = envOrNone(name) orElse alt
 
   // for values based on propFilename
-  def scalaPropOrElse(name: String, alt: String): String = scalaProps.getProperty(name, alt)
+  def scalaPropOrElse(name: String, alt: String): String = scalaPropOrNone(name).getOrElse(alt)
   def scalaPropOrEmpty(name: String): String             = scalaPropOrElse(name, "")
-  def scalaPropOrNone(name: String): Option[String]      = Option(scalaProps.getProperty(name))
+  def scalaPropOrNone(name: String): Option[String]      = Option(scalaProps.getProperty(name)).orElse(Option(System.getProperty("scala." + name)))
```
Comment (retronym) [inline src/library/scala/util/Properties.scala]: `orElse(propOrNone("scala." + name))`

File: src/library/scala/util/Properties.scala [hunk 2]
```diff
@@ -63,9 +63,9 @@ private[scala] trait PropertiesTrait {
   def envOrSome(name: String, alt: Option[String])       = envOrNone(name) orElse alt
 
   // for values based on propFilename
```
Comment (retronym) [inline src/library/scala/util/Properties.scala]: Could you please document the fallback mechanism here?

**Comment by puffnfresh** [comment]
@retronym pushed an updated commit.

**Comment by retronym** [comment]
LGTM

**Pull request merged**